### ISSUE TITLE
Why no fromFile(s: String) instead of fromFileString(s: String)?

### ISSUE THREAD

**Issue by wspringer** [jesseeichar/scala-io#14]
Just a question. Not necessarily an issue. Other than it might be a good idea to document the rationale.

**Comment by jesseeichar** [comment]
This is caused by a technical limitation of Scala.  Right now the signature of fromFile is:

fromFile(File)(implicit CloseAction = Noop)

both curried methods and methods with defaults cannot be overloaded.  So I could remove the CloseAction parameter from Resource methods and then we could have overloaded methods.  I am tempted to do that since one can add CloseActions to Resource objects.

**Comment by wspringer** [comment]
Bummer. What about having a second implicit parameter that basically encodes the policy of turning whatever is passed in as a first argument into a `File` object?

```
fromFile[T](file: T)(implicit action: CloseAction = Noop, 
 strategy: FileCreationStrategy[T])
```

... and then have two implicit objects. One turning `java.io.File` objects into (doh) `java.io.File` objects, and the other turning `String`s into `java.io.File` objects.

**Comment by jesseeichar** [comment]
If I solve:

https://github.com/jesseeichar/scala-io/issues/#issue/18

Then we can fix this issue.  And as you can see in 18 I am leaning towards doing that because I think the parameters add complexity to the signature without enough benefit.  I would love your opinion on that issue as well.

**Comment by jesseeichar** [comment]
closed as of bc5083ce224b1db3cc7fd307541c139770d987b6 (2.9) and 5b2b377ccda4f8105426fffc205b7a46388060e3 (master)
